sidekick/pubspec.lock

```diff
@@ -470,6 +470,13 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "0.7.1"
+  sidekick_plugin_installer:
```

Comment:
This seems wrong, `sidekick_plugin_installer` is not added as dependency

Reply:
The `lint.yaml` workflow runs `dart pub get` and `dart analyze` in the `sidekick` directory. 
This installs the dependencies of `sidekick` and runs static code analysis on all files in the `sidekick` directory.

The problem is that only the dependencies of `sidekick` are installed while static code analysis is run on `sidekick` itself as well as other packages in the `sidekick` directory.

Thus, the workflow returns the following error because the package `sidekick/test/templates/minimal_sidekick_plugin` depends on `sidekick_plugin_installer`. 

```
 error • sidekick/test/templates/minimal_sidekick_plugin/tool/install.dart:3:8 • Target of URI doesn't exist: 'package:sidekick_plugin_installer/sidekick_plugin_installer.dart'. Try creating the file referenced by the
          URI, or Try using a URI for a file that does exist. • uri_does_not_exist
```

To quickly fix this , I added `sidekick_plugin_installer` as a dev_dependency to `sidekick`. There is a comment in `sidekick/pubspec.yaml` explaining this.

An alternative would be to instead modify `.github/workflows/lint.yaml` so that `dart pub get` is not only executed in `sidekick`, but in all packages inside of `sidekick` as well.
Or to run `dart pub get` recursively for all packages inside of the current directory - however I'm not aware of an easy option to do this.

Do you think I should remove the dev_dependency and change the `lint` workflow instead?

Reply:
Let's ignore analysis for the templates 😇 21d41c0c0c0d4483313987db7de5050b8fd2df24